src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs

```diff
@@ -170,5 +172,40 @@ private void AddRequestHeadersAndSetRequestUri(HttpRequestMessage httpRequestMes
             httpRequestMessage.Headers.Add(HttpWorkerConstants.HostVersionHeaderName, ScriptHost.Version);
             httpRequestMessage.Headers.UserAgent.ParseAdd($"{HttpWorkerConstants.UserAgentHeaderValue}/{ScriptHost.Version}");
         }
+
+        public async Task<bool> IsWorkerReady()
+        {
+            await Utility.DelayAsync(WorkerConstants.WorkerInitTimeoutSeconds, WorkerConstants.PollingIntervalMilliseconds, async () =>
+            {
+                return await ShouldContinueWaitingForWorker();
+            });
+            bool continueWaitingForWorker = await ShouldContinueWaitingForWorker();
+            _logger.LogDebug("Is Http Worker Ready:{workerRead}", !continueWaitingForWorker);
+            return !continueWaitingForWorker;
+        }
+
+        private async Task<bool> ShouldContinueWaitingForWorker()
```

Comment:
I think a name like `IsWorkerReady`/`IsWorkerReadyForRequest` would be a bit better and more consistent with other existing checks.